code/obj/machinery/manufacturer.dm

```diff
@@ -1026,6 +1039,30 @@
 					user.u_equip(W)
 					W.dropped()
 
+		else if (istype(W,/obj/item/disk/data/floppy))
+			if (src.manudrive)
+				boutput(user, "<span class='alert'>You swap out the manudrive in the manufacturer with a different one.</span>")
+				src.drive_recipes = null // redundancies sake
+				if (src.manudrive)
+					usr.put_in_hand_or_drop(manudrive)
```

Comment:
considering this code is identical to the manudrive ejection, I'd rather you just make an `eject_manudrive` (or similar name) proc which takes the user as an argument and does these 3 lines.
Then just call that here and in the manudrive ejection via UI.

Reply:
also this should be `user`, not `usr`. They're usually the same but usr leads to  fuckery and shouldn't be used more than necessary.